### PR TITLE
Add ThresholdMarkers controls

### LINKED ISSUE #14293: Indicate selected alerts thresholds on data visualization for query
As user adds threshold rules to their alert check, pass threshold values and ranges to as part of giraffe visualization config, so that thresholds are visible on query visualization. 
Dragging threshold lines up and down should change alert check threshold parameters.

### PR DESCRIPTION
**Pull request by chnn**
Closes #14293 

Introduces a new visualization type (`CheckPlot`) for visualizing checks.

These visualizations are simple `_value` vs. `_time` plots with additional features layered on top. For now, the first of such features is an interactive visualization of check thresholds (`ThresholdMarkers`).

A later PR will add the ability to create a `CheckPlot` through the UI.

dragging crit range

I also changed some confusing naming:

- `XYContainer` -> `XYPlot`
- `HistogramContainer` -> `HistogramPlot`
- `ScatterContainer` -> `ScatterPlot`
- `HeatmapContainer` -> `HeatmapPlot` 

## Screenshots

A `range` threshold with an `OK` status level:

within true ok

A `range` threshold with a `WARN` status level and `within: false`:

within false warn

`INFO` status level:

info status level

`UNKNOWN` status level:

unknown status level

### CODE REVIEW

**Comment by ebb-tide** [comment]
It looks really good!

File: ui/src/shared/components/RangeThresholdMarkers.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+const RangeThresholdMarkers: FunctionComponent<Props> = ({
+  yScale,
+  yDomain,
+  threshold: {level, within, minValue, maxValue},
+  onChangeMinPos,
+  onChangeMaxPos,
+}) => {
+  const minY = clamp(yScale(minValue), yDomain)
+  const maxY = clamp(yScale(maxValue), yDomain)
```
Comment (chnn) [inline ui/src/shared/components/RangeThresholdMarkers.tsx]: oops, just realized this should be 
```suggestion
  const maxY = yScale(clamp(maxValue, yDomain))
```

will fix before merging

File: ui/src/shared/utils/useDragBehavior.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import * as React from 'react'
+import {useCallback} from 'react'
+
+export interface DragEvent {
```
Comment (ebb-tide) [inline ui/src/shared/utils/useDragBehavior.ts]: suprised this type doesn't already exist?

File: ui/src/shared/components/GreaterThresholdMarker.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    <>
+      {value <= yDomain[1] && (
+        <ThresholdMarkerArea
+          level={level}
+          top={yScale(yDomain[1])}
+          height={y - yScale(yDomain[1])}
+        />
+      )}
+      {isInDomain(value, yDomain) && (
+        <ThresholdMarker level={level} y={y} onDrag={onChangePos} />
```
Comment (ebb-tide) [inline ui/src/shared/components/GreaterThresholdMarker.tsx]: soo good!

File: ui/src/shared/components/RangeThresholdMarkers.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          {minValue >= yDomain[0] && (
+            <ThresholdMarkerArea
+              level={level}
+              top={minY}
+              height={yScale(yDomain[0]) - minY}
+            />
+          )}
+        </>
+      )}
+      {isInDomain(minValue, yDomain) && (
```
Comment (ebb-tide) [inline ui/src/shared/components/RangeThresholdMarkers.tsx]: I stared at this for a while before parsing that this was outside the ternary, can the markers be above the marker areas please?

File: ui/src/types/dashboards.ts [hunk 5]
```diff
@@ -286,6 +287,47 @@ export interface ScatterView {
   showNoteWhenEmpty: boolean
 }
 
+export type CheckStatusLevel = 'OK' | 'INFO' | 'WARN' | 'CRIT' | 'UNKNOWN'
```
[the remark this answers is not in the document]
Comment (chnn) [reply ui/src/types/dashboards.ts]: Awesome I will keep my eye out for them

File: ui/src/shared/utils/useDragBehavior.ts (hunk 2 above)
Comment (chnn) [reply ui/src/shared/utils/useDragBehavior.ts]: Hmm we have something like it in giraffe (though incompatible). Maybe that's what you're thinking of?

File: ui/src/shared/components/RangeThresholdMarkers.tsx (hunk 4 above)
Comment (chnn) [reply ui/src/shared/components/RangeThresholdMarkers.tsx]: jeez, almost like you're a human instead of jsx parser! I'll update it

**Pull request merged**